alert-common/src/main/java/com/synopsys/integration/alert/common/descriptor/action/DescriptorActionApi.java

```diff
@@ -49,21 +51,45 @@
         for (final Map.Entry<String, ConfigField> fieldEntry : descriptorFields.entrySet()) {
-            final String fieldKey = fieldEntry.getKey();
-            final Optional<FieldValueModel> optionalField = fieldModel.getField(fieldKey);
-            if (fieldEntry.getValue().isRequired() && optionalField.isEmpty()) {
-                fieldErrors.put(fieldKey, ConfigField.REQUIRED_FIELD_MISSING);
+            final String key = fieldEntry.getKey();
+            final ConfigField field = fieldEntry.getValue();
+            final Optional<FieldValueModel> optionalFieldValue = fieldModel.getField(key);
+            if (field.isRequired() && optionalFieldValue.isEmpty()) {
+                fieldErrors.put(key, ConfigField.REQUIRED_FIELD_MISSING);
             }
-            // field is present now validate the field
-            if (!fieldErrors.containsKey(fieldKey) && optionalField.isPresent()) {
-                final Collection<String> validationErrors = fieldEntry.getValue().validate(optionalField.get(), fieldModel);
+
+            if (!fieldErrors.containsKey(key) && optionalFieldValue.isPresent()) {
+                // field is present now validate the field
+                final FieldValueModel fieldValueModel = optionalFieldValue.get();
+                if (shouldValidateField(fieldValueModel, field.getType())) {
+                    final Set<String> requiredRelatedFields = field.getRequiredRelatedFields();
+                    for (final String relatedFieldKey : requiredRelatedFields) {
+                        final ConfigField relatedField = descriptorFields.get(relatedFieldKey);
+                        validateRelatedFields(relatedField, fieldModel, fieldErrors);
+                    }
+                }
+                final Collection<String> validationErrors = field.validate(fieldValueModel, fieldModel);
                 if (!validationErrors.isEmpty()) {
-                    fieldErrors.put(fieldKey, StringUtils.join(validationErrors, ", "));
+                    fieldErrors.put(key, StringUtils.join(validationErrors, ", "));
                 }
             }
         }
     }
 
+    public Boolean shouldValidateField(final FieldValueModel fieldValueModel, final String type) {
+        final Boolean isValueTrue = fieldValueModel.getValue().map(Boolean::parseBoolean).orElse(false);
+        final Boolean isCheckbox = FieldType.CHECKBOX_INPUT.getFieldTypeName().equals(type);
+        return (isValueTrue && isCheckbox) || (!fieldValueModel.containsNoData() && !isCheckbox);
+    }
+
+    public void validateRelatedFields(final ConfigField field, final FieldModel fieldModel, final Map<String, String> fieldErrors) {
+        final String key = field.getKey();
+        final Optional<FieldValueModel> optionalFieldValue = fieldModel.getField(key);
+        if (optionalFieldValue.isEmpty() || (optionalFieldValue.isPresent() && optionalFieldValue.get().containsNoData())) {
```

Comment:
we are making sure that related fields are set here and not calling validate on the related fields.  Does it make sense to change the method name to validateAnyRelatedFieldsMissing?